### PR TITLE
V030

### PR DESCRIPTION
**Pull request by DisturbedII**
added #13 
added #3

### CODE REVIEW

File: SalienBot/Program.cs [hunk 1]
```diff
@@ -104,67 +104,41 @@ static void Main(string[] args)
...
-                        PRIORITIES.AddRange(prios);
-                    }
-
-                }
-
+                File.WriteAllText("access_token.txt", ACCESS_TOKEN);
+            }
+            else
+            {
+                File.Create("access_token.txt");
```
Comment (StillGreen-san) [inline SalienBot/Program.cs:136]: File write functions should have an option to create the file if it doesn't exist.
No need for if()

File: SalienBot/Program.cs [hunk 2]
```diff
@@ -104,67 +104,41 @@ static void Main(string[] args)
...
             }
-            catch (Exception e) { }
             if (ACCESS_TOKEN == null || ACCESS_TOKEN == "")
             {
                 Console.WriteLine("access_token file empty?");
                 Console.WriteLine("Please get a token from here: https://steamcommunity.com/saliengame/gettoken");
                 Console.WriteLine("Paste token down here:");
                 ACCESS_TOKEN = Console.ReadLine();
             }
-            ACCESS_TOKEN.Trim('"', ' ', ',');
```
Comment (StillGreen-san) [inline SalienBot/Program.cs:117]: Had moved the .Trimm() part down here to catch both file and console input.

File: SalienBot/Program.cs [hunk 3]
```diff
@@ -104,67 +104,41 @@ static void Main(string[] args)
             try
             {
                 string line = File.ReadAllLines("access_token.txt")[0];
-                ACCESS_TOKEN = line;
+                ACCESS_TOKEN = line.Trim(' ','"',',','.');
```
Comment (StillGreen-san) [inline SalienBot/Program.cs]: .Trimm() should be moved down. See comment below.

File: SalienBot/Program.cs [hunk 4]
```diff
@@ -371,6 +347,189 @@ public static void RefreshData()
...
+                    List<Priority> prios = new List<Priority>();
+
+                    int i = 0;
+                    foreach (string Line in lines)
+                    {
+                        i++;
+                        if (Line.Length >= 7)
+                        {
+                            string[] split = Line.Split(':');
+                            if (split[0] == "ROUND_TIME")
```
Comment (StillGreen-san) [inline SalienBot/Program.cs]: Using Switch recommended, if only for the cleaner looks.

File: SalienBot/Program.cs [hunk 5]
```diff
@@ -371,6 +347,189 @@ public static void RefreshData()
...
+            catch (Exception e)
+            {
+                Console.ForegroundColor = ConsoleColor.DarkRed;
+                Console.WriteLine("unexpected error occured in getMainConfig " + e );
+                Console.ResetColor();
+                ExceptionHandling(e);
+            }
+        }
+
+       /* public static void refreshPrioritiesConfig()
```
Comment (StillGreen-san) [inline SalienBot/Program.cs]: Can be removed if the new logic works.

File: SalienBot/Program.cs [hunk 6]
```diff
@@ -504,13 +663,19 @@ public static PlayerInfo GetPlayerInfo()
...
+            }
 
             try
             {
                 pi.clanid = (int)response["clan_info"]["accountid"];
             }
-            catch (Exception e) { }
+            catch (Exception e)
+            {
+                ExceptionHandling(e);
```
Comment (StillGreen-san) [inline SalienBot/Program.cs]: ExceptionHandling is currently not designed to handle those exceptions. As it automatically prints countdown for retry, but here is nothing to retry.
Also for this one, there is no handling required as the value might be empty, which isn't an error.

File: SalienBot/Program.cs [hunk 7]
```diff
@@ -562,6 +727,7 @@ public static void ExceptionHandling(Exception exception)
             {
                 Console.WriteLine("Exception while writing log file:");
                 Console.WriteLine(exception.Message);
+                Console.WriteLine(e);
```
Comment (StillGreen-san) [inline SalienBot/Program.cs:660]: Have you tried this? Not sure if .ToString() is needed for this one.

**Review by StillGreen-san — CHANGES_REQUESTED** [review]
some more work to be done

**Comment by DisturbedII** [comment]
fixes are comitted

**Review by StillGreen-san — APPROVED** [review]

**Pull request merged**